packages/core/src/components/ModalNew/ModalHeader/ModalHeader.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { forwardRef } from "react";
+import cx from "classnames";
+import { getTestId } from "../../../tests/test-ids-utils";
+import { ComponentDefaultTestId } from "../../../tests/constants";
+import styles from "./ModalHeader.module.scss";
+import { ModalHeaderProps } from "./ModalHeader.types";
+import Flex from "../../Flex/Flex";
+import Heading from "../../Heading/Heading";
+import Text from "../../Text/Text";
+import Icon from "../../Icon/Icon";
+
+const ModalHeader = forwardRef(
+  (
+    { title, description, descriptionIcon, className, id, "data-testid": dataTestId }: ModalHeaderProps,
+    ref: React.ForwardedRef<HTMLDivElement>
+  ) => {
+    return (
+      <Flex
+        gap={Flex.gaps.XS}
+        direction={Flex.directions.COLUMN}
+        align={Flex.align.START}
+        ref={ref}
+        className={cx(styles.header, className)}
+        id={id}
+        data-testid={dataTestId || getTestId(ComponentDefaultTestId.MODAL_NEXT_HEADER, id)}
+      >
+        <Heading type={Heading.types.H2} weight={Heading.weights.MEDIUM} maxLines={2} className={styles.title}>
+          {title}
+        </Heading>
+        {description && (
+          <Flex>
+            {descriptionIcon && (
+              <Icon
+                icon={typeof descriptionIcon === "object" ? descriptionIcon.name : descriptionIcon}
+                className={cx(styles.descriptionIcon, typeof descriptionIcon === "object" && descriptionIcon.className)}
+              />
+            )}
+            {typeof description === "string" ? (
+              <Text element="span" type={Text.types.TEXT1}>
```

Comment:
Just wondering why this needs to be a span if it is inside flex

Reply:
Hmm good point, it wasn't initially a flex, but then I thought that people can pass node which would be a block and it would break the inline with the icon.
Do you think I should change it or leave it as is?

Reply:
It is not critical, so whatever you prefer :)

Reply:
Cool, if I'll see it creates an issue for some reason, I'll remove it.
Thanks!

Reply:
Cool, if I'll see it creates an issue for some reason, I'll remove it.
Thanks!